ui/src/app/pages/access/access-renewal.tsx

```diff
@@ -195,28 +193,32 @@ export const AccessRenewal = fp.flow(
   const [loading, setLoading] = useState(false);
   const [, navigateByUrl] = useNavigation();
 
+  const expirableModules = modules.filter(moduleStatus => accessRenewalModules.includes(moduleStatus.moduleName));
+
+  const completeOrBypassed = moduleName => {
+    const status = modules.find(m => m.moduleName === moduleName);
+    const wasBypassed = !!status.bypassEpochMillis;
+    return wasBypassed || !isExpiring(status.expirationEpochMillis);
+  }
 
-  // onMount - as we move between pages, let's make sure we have the latest profile
+  const allModulesCompleteOrBypassed = fp.flow(fp.map('moduleName'), fp.all(completeOrBypassed))(expirableModules);
+
+  // onMount - as we move between pages, let's make sure we have the latest profile and external module information
   useEffect(() => {
     const getProfile = async() => {
       setLoading(true);
       await reloadProfile();
       setLoading(false);
     };
+    const expiringModules = expirableModules
+      .filter(status => status.expirationEpochMillis && isExpiring(status.expirationEpochMillis))
+      .map(status => status.moduleName);
 
+    externalSyncModules(expiringModules);
     getProfile();
```

Comment:
Existing issue, but these two calls are async, yet we immediately call `hideSpinner`. Is that working as intended? Is this handled in a different way on the page already?

Reply:
updated this logic (here and DAR) ... I'll test locally again as well

Reply:
The network timing diagram shows that we now wait for sync-training before refreshing the profile.